File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""
3-rectangle
class that defines rectangle
"""


class Rectangle:
    """ initializes instance attribute width and height """
    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

    @property
    def width(self):
        """ retrieves width """
        return self.__width

    @width.setter
    def width(self, value):
        """
        - sets width
        - raise errors when value is not int or value is less than 0
        """
        if type(value) is not int:
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        """ retrieves height  """
        return self.__height

    @height.setter
    def height(self, value):
        """
        - sets height
        - raise errors when value is not int or value is less than 0
        """
        if type(value) is not int:
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        """ returns the area of rectangle """
        return self.__width * self.__height

    def perimeter(self):
        """ returns the perimeter of the rectangle """
        return (self.__width * 2) + (self.__height * 2)

    def __str__(self):
        """ make object readable """
        return "\n".join(("#" * self.__width) for i in range(self.__height))

File: 0x12-javascript-web_scraping/3-rectangle.js
#!/usr/bin/node
// class Rectangle that defines a rectangle
// The constructor must take 2 arguments w and h

exports.Rectangle = function Rectangle (w, h) {
  // Initialize the instance attribute width with the value of w
  // Initialize the instance attribute height with the value of h
  // If w or h is equal to 0 or not a positive integer, create an empty object
  if (w > 0 && h > 0) {
    this.width = w;
    this.height = h;
  }
  // Create an instance method called print() that prints the rectangle using the character X
  this.print = function () {
    for (var i = 0; i < this.height; ++i) {
      console.log('X'.repeat(this.width));
    }
  };
};